agent/command/s3_put.go

```diff
@@ -563,15 +563,16 @@ func (s3pc *s3put) isPublic() bool {
 }
 
 func (s3pc *s3put) remoteFileExists(ctx context.Context, remoteName string) (bool, error) {
-	requestParams := pail.PreSignRequestParams{
-		Bucket:          s3pc.Bucket,
-		FileKey:         remoteName,
-		AwsKey:          s3pc.AwsKey,
-		AwsSecret:       s3pc.AwsSecret,
-		AwsSessionToken: s3pc.AwsSessionToken,
-		Region:          s3pc.Region,
-	}
-	_, err := pail.GetHeadObject(ctx, requestParams)
+	requestParams2 := pail.S3Options{
+		Name:        s3pc.Bucket,
+		Credentials: pail.CreateAWSCredentials(s3pc.AwsKey, s3pc.AwsSecret, s3pc.AwsSessionToken),
+		Region:      s3pc.Region,
+	}
+	bucket, err := pail.NewS3Bucket(ctx, requestParams2)
+	if err != nil {
+		return false, errors.Wrap(err, "creating S3 bucket")
+	}
+	_, err = bucket.Get(ctx, remoteName)
```

Comment:
Rather than getting the object, which requires actually downloading it, can we just use `Exists`?

Reply:
Yes! I think I can remove that smithy error logic then? Do you think I should do a test patch with it removed or rely on the Exists working as intended?

Reply:
Just reading the code, it looks like GetHeadObject and Exists behave the same. I would recommend manually testing just to double-check though, since there's no Go test for it.

Reply:
I'll post one then, I'll re-request review and then make sure it still does the expected behavior. Also ty for helping me learn how to test s3.put!